Image-builder documentation changes (#10198)
aws/eks-anywhere: docs/content/en/docs/osmgmt/artifacts.md

#### vSphere requirements

image-builder supports both Hashicorp [vsphere-iso](https://developer.hashicorp.com/packer/integrations/hashicorp/vsphere/latest/components/builder/vsphere-iso) and [vsphere-clone](https://developer.hashicorp.com/packer/integrations/hashicorp/vsphere/latest/components/builder/vsphere-clone) Packer Builder for building vSphere OVAs.

##### Permissions

Configure a user with a role containing the following permissions.

The role can be configured programmatically with the `govc` command below, or configured in the vSphere UI using the table below as reference.

> >**_NOTE:_** It is recommended to configure the role at Global Permission level, however, if a user has decided to use specific datastore, network, or resource pool for the image-build process, the role can also be configured with Object-Level Permission. You need to ensure that after configuring object-level permission, the vSphere user being used by image-builder has all the required prvilege on these vSphere object.

Reviewer: ```suggestion
>**_NOTE:_** It is recommended to configure the role at Global Permission level. However, if a user has decided to use a specific datastore, network, or resource pool for the image-build process, the role can also be configured with Object-Level Permission. You need to ensure that after configuring object-level permission, the vSphere user being used by image-builder has all the required privilege on these vSphere object.
```

###### Create role with object-level Permission
A role with object-level permission has to be created through vSphere UI, it is recommended to refer to Broadcom's documentation for this process.

For vSphere 7.0: [Using vCenter Server Roles to Assign Privileges](https://techdocs.broadcom.com/us/en/vmware-cis/vsphere/vsphere/7-0/vsphere-security-7-0/vsphere-permissions-and-user-management-tasks/using-roles-to-assign-privileges.html)

For vSphere 8.0: [Using vCenter Server Roles to Assign Privileges](https://techdocs.broadcom.com/us/en/vmware-cis/vsphere/vsphere/8-0/vsphere-security-8-0/vsphere-permissions-and-user-management-tasks/using-roles-to-assign-privileges.html)

Example of creating role with object-level permission on a Datastore in vSphere 8.0:

1. Navigate to `Administration > Single Sign On > Users and Groups`, create a user that will be used to authenticate to the vSphere server.

> 2. Create a role and configure its permission following the table in `Create role with global permission` section.

Reviewer: ```suggestion
2. Create a role and configure its permission following the table in the `Create role with global permission` section.
```

> 3. Navigate to `Inventory > Storage`, identify the datastore that will be used in the image-build process.

Reviewer: ```suggestion
3. Navigate to `Inventory > Storage` and identify the datastore that will be used in the image-build process.
```

4. Right click on the datastore, and click `Add Permissions`, select the user created in step 1 and the role created in step 2.

> Object-level permission can be configured on `Network`, `Datastore` and `Resource Pool` object.

Reviewer: ```suggestion
Object-level permission can be configured on `Network`, `Datastore` and `Resource Pool` objects.
```

   ##### **[resource_pool](https://developer.hashicorp.com/packer/integrations/hashicorp/vsphere/latest/components/builder/vsphere-iso#connection-configuration)**
   The vSphere resource pool where the virtual machine is created. If this is not specified, the root resource pool associated with the cluster is used.

   ##### **[template](https://developer.hashicorp.com/packer/integrations/hashicorp/vsphere/latest/components/builder/vsphere-clone#clone-configuration)**

>    The name of the source virtual machine/OVF template to clone(required if using vsphere-clone builder).

Reviewer: ```suggestion
   The name of the source virtual machine/OVF template to clone (required if using vsphere-clone builder).
```

     "rhel_password": "<RHSM password>"
   }
   ```

>    >**_NOTE:_** Image-builder currently only supports building RHEL images with full DVD ISO and does not support minimal ISO.

Reviewer: ```suggestion
   >**_NOTE:_** Image-builder currently only supports building RHEL images with the full DVD ISO and does not support minimal ISOs.
```

   To create a RHEL-based image, run `image-builder` with the following options:

      * `--os`: `redhat`
      * `--os-version`: `8`, `9` (default: `8`)
      * `--hypervisor`: For vSphere use `vsphere`
      * `--release-channel`: Supported Kubernetes versions, from 1-28 to 1-34.
      * `--vsphere-config`: vSphere configuration file (`vsphere.json` in this example)

      ```bash
      image-builder build --os redhat --hypervisor vsphere --release-channel 1-31 --vsphere-config vsphere.json
      ```

2. Create an Ubuntu or Redhat image with vsphere-clone builder:

>    [Vsphere clone builder](https://developer.hashicorp.com/packer/integrations/hashicorp/vsphere/latest/components/builder/vsphere-clone) support has been added to EKS-A image-builder in v0.7.0. The vsphere clone builder allows user to clone an existing virtual machine/OVF template on vCenter and convert it into an EKS-A node image.

Reviewer: ```suggestion
   [Vsphere clone builder](https://developer.hashicorp.com/packer/integrations/hashicorp/vsphere/latest/components/builder/vsphere-clone) support has been added to EKS-A image-builder in v0.7.0. The vsphere clone builder allows user to clone an existing virtual machine/OVF template on vCenter and convert it into an EKS Anywhere node image.
```
I was told by marketing a long time ago to always spell out "EKS Anywhere", and only use EKS-A, eksa, or eks-a if it is literally used as an option or part of a URL, or similar.

   **Requirements on base template**
   The base template needs to have an SSH user named `builder` with a known password. This is because image-builder will be using this user to establish SSH connection to the builder machine on vSphere.

>    The password needs to be set via environment variable

Reviewer: ```suggestion
   The password needs to be set via the following environment variable:
```

            <td>22.04.3</td>
            <td>22.04</td>
        </tr>
        <tr>
            <td>24.04.1</td>
            <td>24.04</td>
            <td rowspan=2>Supported on Baremetal and vSphere</td>
        </tr>
        <tr >
            <td rowspan=2><b>RHEL</b></td>
            <td>8.10</td>
            <td>8</td>
            <td rowspan=2>8</td>
            <td>All hypervisors except AMI</td>
        </tr>
        <tr>
            <td>9.6</td>
            <td>9</td>
            <td>All hypervisors except AMI</td>
        </tr>
    </tbody>
</table>

> ### Configuring Ansible Logging Verbosity

Reviewer: ```suggestion
### Configuring Ansible logging verbosity
```

> By default `image-builder` does not display the details of the Ansible tasks during image-build process, but it allows user to specify logging verbosity for Ansible using `--ansible-verbosity` option, the value is in range `0-6`.

Reviewer: ```suggestion
By default `image-builder` does not display the details of the Ansible tasks during the image-build process. But it allows users to specify logging verbosity for Ansible using the `--ansible-verbosity` option. The value is in range `0-6`.
```

   image-builder build --os <OS> --hypervisor <hypervisor> --release-channel <release channel> --<hypervisor>-config config.json
   ```

### Red Hat Image Minor Version Upgrade Behavior

> By default, `image-builder` performs a `yum update` task during the image-build process, which will updates the RHEL minor version to the latest. For example, if a user builds a RHEL image using an ISO of version `9.4`, the final image artifact will have RHEL version `9.6`.

Reviewer: ```suggestion
By default, `image-builder` performs a `yum update` task during the image-build process, which updates the RHEL minor version to the latest. For example, if a user builds a RHEL image using an ISO of version `9.4`, the final image artifact will have RHEL version `9.6`.
```

This is an expected behavior of image-builder. However, if you want to preserve the Red Hat minor version, you can manually specify the `rhsm_server_release_version` in config.json file to keep Red Hat minor version on the value that you specified.

> For example, if user wants the Red Hat minor version to remain `9.4`, add the following to config.json file:

Reviewer: ```suggestion
For example, if you want the Red Hat minor version to remain `9.4`, add the following to config.json file:
```
